library/video_transforms/aligned_edge_detection_process.txx

```diff
@@ -214,6 +214,7 @@ aligned_edge_detection_process<PixType>
     }
     else
     {
+      // FIXME this can't be right; need to copy the data into the view
       combined_edges = joint_nms_edges;
```

Comment:
Previously (before your change) I think this fine was fine because the combined_edges class was being passed by reference, I'm changing it to a copy reformat though since this branch is required to work with 1.65.

Reply:
...except it *wasn't*. This gets called via a `function<Args...>`[[1](https://github.com/Kitware/burn-out/blob/0f2edd405873e64cc0fc1129ba32ba809c3e3332/library/video_transforms/threaded_image_transform.h#L46)]. The `operator()` of that `function` is therefore not receiving references, it is receiving copies. `process_region` may be receiving a reference, but it is a reference to an internal member of the `function`, which is effectively a dead reference, and *not* a reference to anything which might be meaningfully modified. (Basically, this was always broken, but older Boost was sloppy about letting you think it wasn't.)

Moreover, even if this was not the case, the `function` would be holding references to [these](https://github.com/Kitware/burn-out/blob/0f2edd405873e64cc0fc1129ba32ba809c3e3332/library/video_transforms/threaded_image_transform.h#L201), which are dead before the `function` gets dispatched. (Thus, changing the `function` to take reference objects would make things worse, since the references would be to *actually* dead objects, not just "effectively dead" objects.)

The only reason this code mostly works is because the objects are glorified smart pointers; they internally retain a reference to memory held elsewhere, which is (I guess/hope) still "live". (Note: there is nothing necessarily wrong with that design, as long as its operation is correctly understood.) This particular statement however is broken because it reassigns that internal reference, rather than writing back to the underlying buffer. Hence the "FIXME".

(More particularly: this statement says 'reassign this view to reference some other view's data'. What it *needs* to say is 'copy this other view's data *into the data of this view*'. At least, I am guessing that was the intent... It may also be that this statement should just be deleted?)